doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
+This function integrate over the depth dimension. This function does a weighted
+sum along the `z`-coordinate, and removes the `z` direction of the output
+cube. This preprocessor takes no arguments.
 
 See also :func:`esmvalcore.preprocessor.depth_integration`.
 
-4. extract_transect
--------------------
 
-This function extracts data along a line of constant latitude or longitude.
-This function takes two arguments, although only one is strictly required:
-* latitude
-* longitude
+``extract_transect``
+--------------------
 
-One of these arguments needs to be set to a float, and the other can then be
-either ignored or set to a minimum or maximum value. For example, if latitude
-is set to 0 and longitude is left blank, the function would produce a cube
-along the equator. If latitude is set to to 0 and longitude to `[40., 100.]` it
-will produce a transect of the equator in the Indian Ocean.
+This function extract data along a line of constant latitude or longitude.
+This function takes two arguments, although only one is strictly required.
+The two arguments are ``latitude`` and ``longitude``. One of these arguments
+needs to be set to a float, and the other can then be either ignored or set to
+a minimum or maximum value.
+
+For example, if we set latitude to 0 N and leave longitude blank, it would
+produce a cube along the Equator. On the other hand, if we set latitude to 0
+and then set longitude to ``[40., 100.]`` this will produce a transect of the
+Equator in the Indian Ocean.
 
 See also :func:`esmvalcore.preprocessor.extract_transect`.
 
-5. extract_trajectory
----------------------
 
-This function extracts data along a specified trajectory. It requires three
-arguments:
-* latitude_points: list of latitude coordinates
-* longitude_points: list of longiute coordinates
-* number_points: if two points are provided, the `number_points` argument is
-used to set the number of places to extract between the two end points.
+``extract_trajectory``
+----------------------
+
+This function extract data along a specified trajectory.
+The three areguments are: ``latitudes``, ``longitudes`` and number of point
+needed for extrapolation ``number_points``.
+
+If two points are provided, the ``number_points`` argument is used to set a
+the number of places to extract between the two end points.
 
-If more than two points are provided, then extract_trajectory will produce a
-cube which has extrapolated the data of the cube to those points, and
-`number_points` is not needed. Note that this function uses the expensive
-interpolate method, but it may be necessary for irregular grids.
+If more than two points are provided, then ``extract_trajectory`` will produce
+a cube which has extrapolated the data of the cube to those points, and
+``number_points`` is not needed.
+
+Note that this function uses the expensive ``interpolate`` method from
+``Iris.analysis.trajectory``, but it may be necceasiry for irregular grids.
```

Comment:
Should be "necessary" instead of "necceasiry"?

Reply:
Done